toy_vm.py

```diff
@@ -135,21 +146,52 @@ class InstructionMeta:
 GenericOneInlineParamIns = InstructionMeta(params=[InlineParamToken])
 GenericOneLabelOrInlineParamIns = InstructionMeta(params=[LabelOrInlineParamToken])
 
-INSTRUCTIONS_META = {
-    InstructionCodes.NOP: InstructionMeta(),
+
+NOPARAMS_INSTRUCTIONS_META = {
```

Comment:
Here you can kind of see what I was thinking with the instructions and macros meta ... more programmatic way of defining groups of insts. and also the assember not having to be redone for every VM but be able to compline diff syntaxes just based on the 'family' of toke types being fed. The string splitting would also have to be made more customizable of course ... but kind of works as is now for the f4 arch.